### PR TITLE
Fix naming of layers when building with symbolic tensors

### PR DESCRIPTION
**Pull request by kmigdol**
https://github.com/tensorflow/tensorflow/issues/27298

### CODE REVIEW

File: keras/engine/base_layer.py [hunk 1]
```diff
@@ -2199,21 +2199,24 @@ def _infer_output_signature(self, inputs, args, kwargs, input_masks):
...
-        with autocast_variable.enable_auto_cast_variables(
-            self._compute_dtype_object):
-          # Build layer if applicable (if the `build` method has been
-          # overridden).
-          # TODO(kaftan): do we maybe_build here, or have we already done it?
-          self._maybe_build(inputs)
-          inputs = self._maybe_cast_inputs(inputs)
-          outputs = call_fn(inputs, *args, **kwargs)
+      ctx = context.context()
+      assert ctx is not None
```
Comment (fchollet) [inline keras/engine/base_layer.py:2203]: You can remove this statement

File: keras/engine/base_layer.py [hunk 2]
```diff
@@ -2199,21 +2199,24 @@ def _infer_output_signature(self, inputs, args, kwargs, input_masks):
...
+      with ops.name_scope(ctx.scope_name[:-1]):
+        with backend.name_scope(self._name_scope()):  # pylint: disable=not-callable
+          with autocast_variable.enable_auto_cast_variables(
+                  self._compute_dtype_object):
+            # Build layer if applicable (if the `build` method has been
+            # overridden).
+            # TODO(kaftan): do we maybe_build here, or have we already done it?
```
Comment (fchollet) [inline keras/engine/base_layer.py:2210]: You can remove this TODO.

File: keras/engine/base_layer.py [hunk 3]
```diff
@@ -2199,21 +2199,24 @@ def _infer_output_signature(self, inputs, args, kwargs, input_masks):
...
-          # Build layer if applicable (if the `build` method has been
-          # overridden).
-          # TODO(kaftan): do we maybe_build here, or have we already done it?
-          self._maybe_build(inputs)
-          inputs = self._maybe_cast_inputs(inputs)
-          outputs = call_fn(inputs, *args, **kwargs)
+      ctx = context.context()
+      assert ctx is not None
+      with ops.name_scope(ctx.scope_name[:-1]):
+        with backend.name_scope(self._name_scope()):  # pylint: disable=not-callable
```
Comment (fchollet) [inline keras/engine/base_layer.py:2205]: If wonder if stacking these 2 name scopes may have unexpected side effects. Why not make the outer scope part of `self._name_scope`?

**Review by fchollet — COMMENTED** [review]
Thanks for the PR. The target behavior is reasonable, but there's a high likelihood that this will be breaking change, hence we might not be able merge it in order to preserve backwards compatibility.

**Comment by gbaned** [comment]
@kmigdol  Can you please check @fchollet's comments and keep us posted ? Thanks!

**Comment by gbaned** [comment]
@kmigdol Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
@kmigdol Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
Hi @kmigdol I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen! Thank you!

**Pull request closed without merging**